### DISCUSSION TITLE
Use file as search input

### DISCUSSION THREAD

**Discussion by PoussinBleu** [dnGrep/dnGrep#1138] [Q&A] [answered]
Hello,
dnGrep is amazing. I need to find some id's hundreds of archived files in a folder and dnGrep do that really fast. The problem is that I have like 150 id's to search for and doing it one at a time is quite time consuming. I don't find the possibility to use for example a txt file with a list of "Search for " values ? 

And another question would be to know if dnGrep find my file in an archive, is it possible to copy this file instead of the archive itself to my drive ?
Thanks a lot for your precious help.

**Comment by doug24** [comment]
For the list of ids, no there is no search input file feature.  There currently is no way to run a search and add to the current set of results, so you cannot use scripts, either.  I can look at this, but I'm pretty busy right now and it probably will not happen quickly. 

For copying a file from an archive - that is probably the way it *should* work, but as you have seen, does not.  I can look at fixing this, too.

**Comment by doug24** [comment]
One thing you can do now is to use the Boolean operators search option, and search for a group of ids.  It seems fast to search for about 10 at a time, but there is an exponential (2 to the power of N Boolean operators) in the evaluator, so it quickly grows out of control. But doing 15 searches is way better than 150. It would look something like this:

```
1000 OR 1001 OR 1002 OR 1003 OR 1004 OR 1005 OR 1006 OR 1007 OR 1008 OR 1009
```

**Reply by PoussinBleu** [reply]
Thanks a lot for this advise. Effectively, it will help a lot. I just test it and it takes 2,5 minutes for the search of 2 id's instead of 1 minute for 1 Id. I just have now to download the archive and find manually the file in it. I'll wait for your fix then ;-)
Thank you very much !

**Comment by doug24** [comment, marked as answer]
Added in [v4.2.46](https://github.com/dnGrep/dnGrep/releases/tag/v4.2.46.0)

**Reply by PoussinBleu** [reply]
Hello thank a lot, I try it and it works quite well !
If I may give you a small remark, if we don't read your release note specific for this version, we don't know that the "Search for" field can be filled with a URI of a file. Maybe a specific field or just a tooltip that explain it could be usefull ;-)

**Reply by doug24** [reply]
I need to update the user guide, but I just have not had time to do it yet.  There are many features like this, and there is just no way to include them all in tooltips.

**Reply by PoussinBleu** [reply]
I understand, no problem ;-) Good luck then and thanks again for this great app !!
